apps/web/src/services/dashboards/components/dashboard-folder/DashboardFolderDeleteModal.vue

```diff
@@ -0,0 +1,216 @@
+
+import { useDashboardMainPageStore } from '@/services/dashboards/stores/dashboard-main-page-store';
+import type { DashboardTreeDataType } from '@/services/dashboards/types/dashboard-folder-type';
+
+
+
+const DELETE_TABLE_FIELDS = [
+    { name: 'name', label: 'Name' },
+    { name: 'location', label: 'Location' },
+];
+type FolderDeleteParams = PublicFolderDeleteParameters | PrivateFolderDeleteParameters;
+type DashboardDeleteParams = PublicDashboardDeleteParameters | PrivateDashboardDeleteParameters;
+interface DataTableItem {
+    id: string;
+    name: string;
+    location?: string;
+    type: 'DASHBOARD' | 'FOLDER';
+}
+interface Props {
+    visible?: boolean;
+}
+const props = withDefaults(defineProps<Props>(), {
+    visible: false,
+});
+const emit = defineEmits<{(e: 'update:visible', visible: boolean): void,
+}>();
+const dashboardStore = useDashboardStore();
+const dashboardMainPageStore = useDashboardMainPageStore();
+const dashboardMainPageState = dashboardMainPageStore.state;
+const dashboardMainPageGetters = dashboardMainPageStore.getters;
+const state = reactive({
+    loading: false,
+    proxyVisible: useProxyValue<boolean>('visible', props, emit),
+    dashboardTreeData: computed<TreeNode<DashboardTreeDataType>[]>(() => [...dashboardMainPageGetters.publicDashboardTreeData, ...dashboardMainPageGetters.privateDashboardTreeData]),
+    tableItems: computed<DataTableItem[]>(() => {
+        if (dashboardMainPageState.selectedFolderId) return getSelectedFolderItems(dashboardMainPageState.selectedFolderId);
+        return getSelectedFolderAndDashboardItems();
+    }),
+});
+
+/* Util */
+const getSelectedFolderAndDashboardItems = (): DataTableItem[] => {
+    const _tableItems: DataTableItem[] = [];
+    dashboardMainPageGetters.selectedTreeData.forEach((node) => {
+        if (node.data.type === 'FOLDER') {
+            _tableItems.push({
+                id: node.data.id,
+                name: node.data.name,
+                type: 'FOLDER',
+            });
+            node.children?.forEach((child) => {
+                _tableItems.push({
+                    id: child.data.id,
+                    name: child.data.name,
+                    location: node.data.name,
+                    type: 'DASHBOARD',
+                });
+            });
+        } else {
+            const _folderId = node.data?.folderId;
+            const _folderName = state.dashboardTreeData.find((d) => d.id === _folderId)?.data?.name;
+            _tableItems.push({
+                id: node.data.id,
+                name: node.data.name,
+                location: _folderName,
+                type: 'DASHBOARD',
+            });
+        }
+    });
+    return _tableItems;
+};
+const getSelectedFolderItems = (selectedFolderId: string): DataTableItem[] => {
+    const _tableItems: DataTableItem[] = [];
+    const _selectedFolder = state.dashboardTreeData.find((d) => d.id === selectedFolderId);
+    if (!_selectedFolder) return _tableItems;
```

Comment:
nitpick: I think this flow is pretty embarrassing - User selects folder, but if it's absent on tree data, then just show empty table.🥲